File: testdata/jennies/builders/builder_delegation/PHPBuilder/src/BuilderDelegation/DashboardBuilder.php
<?php

namespace Grafana\Foundation\BuilderDelegation;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegation\Dashboard>
 */
class DashboardBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\BuilderDelegation\Dashboard $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\BuilderDelegation\Dashboard();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\BuilderDelegation\Dashboard
     */
    public function build()
    {
        return $this->internal;
    }

    public function id(int $id): static
    {
        $this->internal->id = $id;
    
        return $this;
    }

    public function title(string $title): static
    {
        $this->internal->title = $title;
    
        return $this;
    }

    /**
     * will be expanded to []cog.Builder<DashboardLink>
     * @param array<\Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegation\DashboardLink>> $links
     */
    public function links(array $links): static
    {
            $linksResources = [];
            foreach ($links as $r1) {
                    $linksResources[] = $r1->build();
            }
        $this->internal->links = $linksResources;
    
        return $this;
    }

    /**
     * will be expanded to [][]cog.Builder<DashboardLink>
     * @param array<array<\Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegation\DashboardLink>>> $linksOfLinks
     */
    public function linksOfLinks(array $linksOfLinks): static
    {
            $linksOfLinksResources = [];
            foreach ($linksOfLinks as $r1) {
                    $linksOfLinksDepth1 = [];
            foreach ($r1 as $r2) {
                    $linksOfLinksDepth1[] = $r2->build();
            }
    
                    $linksOfLinksResources[] = $linksOfLinksDepth1;
            }
        $this->internal->linksOfLinks = $linksOfLinksResources;
    
        return $this;
    }

    /**
     * will be expanded to cog.Builder<DashboardLink>
     * @param \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegation\DashboardLink> $singleLink
     */
    public function singleLink(\Grafana\Foundation\Cog\Builder $singleLink): static
    {
        $singleLinkResource = $singleLink->build();
        $this->internal->singleLink = $singleLinkResource;
    
        return $this;
    }

}

File: testdata/jennies/builders/builder_delegation/CSharpBuilder/src/Grafana/Foundation/BuilderDelegation/DashboardBuilder.cs
namespace Grafana.Foundation.BuilderDelegation;


public class DashboardBuilder : Cog.IBuilder<Dashboard>
{
    protected readonly Dashboard @internal;

    public DashboardBuilder()
    {
        this.@internal = new Dashboard();
    }

    public DashboardBuilder Id(long id)
    {
        this.@internal.Id = id;
        return this;
    }

    public DashboardBuilder Title(string title)
    {
        this.@internal.Title = title;
        return this;
    }

    public DashboardBuilder Links(List<Cog.IBuilder<DashboardLink>> links)
    {
        List<DashboardLink> linksResources = new List<DashboardLink>();
        foreach (Cog.IBuilder<DashboardLink> r1 in links)
        {
                DashboardLink linksDepth1 = r1.Build();
                linksResources.Add(linksDepth1);
        }
        this.@internal.Links = linksResources;
        return this;
    }

    public DashboardBuilder LinksOfLinks(List<List<Cog.IBuilder<DashboardLink>>> linksOfLinks)
    {
        List<List<DashboardLink>> linksOfLinksResources = new List<List<DashboardLink>>();
        foreach (List<Cog.IBuilder<DashboardLink>> r1 in linksOfLinks)
        {
                List<DashboardLink> linksOfLinksDepth1 = new List<DashboardLink>();
        foreach (Cog.IBuilder<DashboardLink> r2 in r1)
        {
                DashboardLink linksOfLinksDepth2 = r2.Build();
                linksOfLinksDepth1.Add(linksOfLinksDepth2);
        }

                linksOfLinksResources.Add(linksOfLinksDepth1);
        }
        this.@internal.LinksOfLinks = linksOfLinksResources;
        return this;
    }

    public DashboardBuilder SingleLink(Cog.IBuilder<DashboardLink> singleLink)
    {
        DashboardLink singleLinkResource = singleLink.Build();
        this.@internal.SingleLink = singleLinkResource;
        return this;
    }

    public Dashboard Build()
    {
        return this.@internal;
    }
}
